homeassistant/components/growatt_server/sensor.py

```diff
@@ -504,6 +708,39 @@ def update(self):
                     self.plant_id, self.device_id
                 )
                 self.data = {**storage_info_detail, **storage_energy_overview}
+            elif self.growatt_type == "mix":
+                mix_info = self.api.mix_info(self.device_id)
+                mix_totals = self.api.mix_totals(self.device_id, self.plant_id)
+                mix_system_status = self.api.mix_system_status(
+                    self.device_id, self.plant_id
+                )
+
+                mix_detail = self.api.mix_detail(
+                    self.device_id, self.plant_id, date=datetime.datetime.now()
+                )
+                # Get the chart data and work out the time of the last entry, use this as the last time data was published to the Growatt Server
+                mix_chart_entries = mix_detail["chartData"]
+                sorted_keys = sorted(mix_chart_entries)
+                updated_time = dt.parse_datetime(
```

Comment:
This should be UTC time and isoformatted. Convert it to UTC time and then isoformat it. 

Use this:
https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat

Reply:
Before I push this up, is this what you're looking for:

```
date_now = dt.now().date()
last_updated_time = dt.parse_time(str(sorted_keys[-1]))
combined_timestamp = datetime.datetime.combine(
    date_now, last_updated_time
)
combined_timestamp_utc = dt.as_utc(combined_timestamp)
updated_time = dt.parse_datetime(combined_timestamp_utc.isoformat())
mix_detail["lastdataupdate"] = updated_time
```

Is it even necessary to do the `updated_time = dt.parse_datetime(combined_timestamp_utc.isoformat())` or can we just assign the UTC timestamp to the `mix_detail` entry?

Reply:
We don't need to use `dt.parse_datetime`. `combined_timestamp_utc.isoformat` will return an isoformatted datetime string, which we want.

Reply:
Pushed latest changes